compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1983,7 +2007,31 @@ impl<'context> Elaborator<'context> {
             self.file = trait_impl.file_id;
             self.local_module = trait_impl.module_id;
 
-            let trait_id = self.resolve_trait_by_path(trait_impl.trait_path.clone());
+            let trait_type = match &trait_impl.r#trait.typ {
+                UnresolvedTypeData::Named(trait_path, trait_generics, _) => {
+                    TraitType::Named(trait_path.clone(), trait_generics.clone())
+                }
+                UnresolvedTypeData::Resolved(quoted_type_id) => {
```

Comment:
Do we need to define this new TraitType enum or could we just extract the trait_id, path_span, and trait_generics here instead?

Reply:
Done, thanks! It ends being much clearer and shorter.